magellan-library/src/main/java/com/wealthfront/magellan/lifecycle/CreateAndAttachFieldToLifecycleWhenShownDelegate.kt

```diff
@@ -20,3 +21,40 @@ public class CreateAndAttachFieldToLifecycleWhenShownDelegate<Field>(
 
 public fun <Field> LifecycleOwner.createAndAttachFieldToLifecycleWhenShown(fieldSupplier: (Context) -> Field): AttachFieldToLifecycleDelegate<CreateAndAttachFieldToLifecycleWhenShownDelegate<Field>, Field?> =
   attachFieldToLifecycle(CreateAndAttachFieldToLifecycleWhenShownDelegate(fieldSupplier), { it.field })
+
+
+public class OverrideableShownFieldDelegate<PropertyType>(
```

Comment:
I couldn't use CreateAndAttachFieldToLifecycleWhenShownDelegate because it's built on top of AttachFieldToLifecycleDelegate, which ceases lifecycle listening whenever fields are overridden. Since I am planning to override templatedView (inside NavigatorDelegate), yet *also* want it to stay attached (so that it get torn down onHide, and initialized to a good default inside onShow) I wrote this relatively naive Delegate instead